zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/job/ActivatableJobsPushTest.java

```diff
@@ -277,7 +278,10 @@ private void assertEventOrder(
 
   private void assertActivatedJobsPushed(final Long jobKey, final int activationCount) {
     final var activatedJobs = jobStream.getActivatedJobs();
-    assertThat(activatedJobs).hasSize(activationCount);
+    await("waiting for the expected number of jobs to be activated")
+        .atMost(Duration.ofMillis(300))
+        .pollInterval(Duration.ofMillis(10))
+        .untilAsserted(() -> assertThat(activatedJobs).hasSize(activationCount));
```

Comment:
❓ Does this make a difference? Do we need to pull the `jobStream.getActivatedJobs()` call inside the `await` as well?

Reply:
@tmetzke It's unnecessary as `jobStream.getActivatedJobs()` returns a reference (not a copy) to the activated jobs list.

But I will move `jobStream.getActivatedJobs()` to `. untilAsserted` to avoid confusion.

Thanks for highlighting this!

Reply:
Ah nice, that makes sense, I was hoping for something like this! 👍

Reply:
Solution updated)